### ISSUE TITLE
Regression in linking to external libsemigroups

### ISSUE THREAD

**Issue by dimpase** [semigroups/Semigroups#938]
While `semigroups` still used libtool, it correctly linked external `libsemigroups` using `-rpath` etc. E.g. this is the case for `semigroups` in GAP 4.12.2.

As uncovered in #937, this is no longer the case (see also #928). It will be problematic for e.g. Sage and conda to package in such a state.

**Comment by fingolfin** [comment]
In `NormalizInterface` there is a similar situation: it links against `libnormaliz`, and by default uses a copy of that bundled with `NormalizInterface`. But the user can specify `--with-normal=<path>` to point at another copy of the library. Doing so sets `CPPFLAGS` and `LDFLAGS` suitable -- including setting the rpath. Roughly speaking:
```
CPPFLAGS="-I$path/include"
LDFLAGS="-L$path/lib -Wl,-rpath,$path/lib"
```
With this approach, if libsemigroups is installed in `/usr`, then instead of doing `./configure --with-external-libsemigroups` one would do `./configure --with-libsemigroups=/usr`

That is very simple to implement (I could provide a patch) and so far seems to work everywhere people tried... I am not sure anyone tried on OpenBSD, though -- but perhaps @dimpase could try to see if he can get this to work with `NormalizInterface`. If yes, we could use it for Semigroups, too. If not, well, then we can analyze the problem, try to improve it in `NormalizeInterface`, and repeat...

**Comment by james-d-mitchell** [comment]
@fingolfin Thanks very much for the input!

**Comment by james-d-mitchell** [comment]
@fingolfin @dimpase I think this is still an issue, right? If so, and either of you is able to submit a patch for this, then that'd be greatly appreciated.

**Comment by dimpase** [comment]
I can do it, I think.

I am also very tempted to refactor `m4/ax_check_libsemigroup.m4`, removing unneeded temporary variables used for controlling the program flow (all what is done there may be done by moving code chunks into appropriate `[...]` blocks).

Would you mind if I do the latter, too?

**Comment by james-d-mitchell** [comment]
Not at all! I'd be most happy!

**Comment by james-d-mitchell** [comment]
Resolved in v5.3.0
